delegator.py

```diff
@@ -86,7 +87,10 @@ def std_err(self):
     @property
     def err(self):
         if self._uses_subprocess:
-            return self.std_err.read()
+            if self.__err:
```

Comment:
Sorry, one more quick note @shaded-enmity. I think it may be better to lift the `if self.__err:` block out of the `_uses_subprocess` conditional. If `self.__err` is set, let's just return it like we do in `out()`.